### PR TITLE
Add AmberScript generation for image and compute, and relevant runspv tests

### PR DESCRIPTION
**Pull request by hevrard**
The AmberScript generated is usable, but it still misses features like
embedding the original GLSL as comment.

### CODE REVIEW

**Comment by hevrard** [comment]
Tests and compute shaders are missing, I was slowed down by test issues.

This should be mergeable now, if you don't want to wait for me to add compute shader and tests.

**Comment by hevrard** [comment]
This PR has been completely reworked, it now implements support for AmberScript format on both image and compute. Additional runspv tests have been added.

File: python/src/main/python/drivers/runspv.py [hunk 1]
```diff
@@ -925,13 +958,159 @@ def amberscriptify_image(
...
+    result = ''
+    with open(uniform_json, 'r') as f:
+        j = json.load(f)
+    for name, entry in j.items():
+
+        if name == '$compute':
+            continue
+
+        func = entry['func']
+        if func not in UNIFORM_TYPE.keys():
```
Comment (afd) [inline python/src/main/python/drivers/runspv.py]: We should probably change the following to "raise"; @paulthomson what do you think?

File: python/src/main/python/drivers/runspv.py [hunk 2]
```diff
@@ -925,13 +958,159 @@ def amberscriptify_image(
...
+    else:
+        result += 'SHADER vertex gfz_vert PASSTHROUGH\n\n'
+
+    result += 'SHADER fragment gfz_frag SPIRV-ASM\n'
+    result += spv_get_disassembly(frag)
+    result += 'END\n\n'
+
+    # This buffer MUST be named framebuffer to be able to retrieve the image
+    # Format MUST be B8G8R8A8_UNORM (other options may become available once
+    # Amber supports more format for image extraction)
```
Comment (afd) [inline python/src/main/python/drivers/runspv.py]: ^ format -> formats

**Pull request merged**